File: JLBannerViewDemo/JLBannerViewDemo/CastBannerCell.m
//
//  CollectionViewCell.m
//  JLBannerViewDemo
//
//  Created by Jangsy7883 on 2017. 4. 26..
//  Copyright © 2017년 Studio. All rights reserved.
//

#import "CastBannerCell.h"

@implementation CastBannerCell

- (void)layoutSubviews {
    self.bannerView.frame = self.contentView.bounds;
    
    if (self.bannerView.superview == nil) {
        [self.contentView addSubview:self.bannerView];
        [self.bannerView reloadData];
    }
}

- (JLBannerView *)bannerView {
    if (!_bannerView) {
        _bannerView = [[JLBannerView alloc] init];
    }
    return _bannerView;
}
@end

File: JLBannerViewDemo/JLBannerViewDemo/CastBannerCell.swift
//
//  CastBannerCell.swift
//  JLBannerViewDemo
//
//  Created by Woody on 2017. 5. 25..
//  Copyright © 2017년 Woody. All rights reserved.
//

import UIKit

class CastBannerCell: UICollectionViewCell {
    var _bannerView: JLBannerView? = nil
    var bannerView: JLBannerView? {
        if _bannerView == nil {
            _bannerView = JLBannerView()
        }
        return _bannerView
    }

    override func layoutSubviews() {
        super.layoutSubviews()

        bannerView?.frame = bounds
        if bannerView?.superview == nil {
            addSubview(bannerView!)
            bannerView?.reloadData()
        }
    }
}
